File: app/src/main/java/com/example/user/biblejournal/view/NoteListFragment.java
package com.example.user.biblejournal.view;

import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.example.user.biblejournal.R;
import com.example.user.biblejournal.adapter.NoteListAdapter;
import com.example.user.biblejournal.model.database.NoteEntity;
import com.example.user.biblejournal.viewmodel.NoteViewModel;
import com.google.android.material.floatingactionbutton.FloatingActionButton;

import java.util.List;
import java.util.Objects;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProviders;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;


public class NoteListFragment extends Fragment implements NoteListAdapter.ListItemListener {
    private NoteListListener noteListListener;
    private FloatingActionButton floatingActionButton;
    private NoteViewModel noteViewModel;
    private RecyclerView noteListRecyclerView;

    public static NoteListFragment newInstance() {
        return new NoteListFragment();
    }

    public interface NoteListListener {
        void startNewNote();
        void editNote(int noteId);
    }

    @Override
    public void onAttach(@NonNull Context context) {
        super.onAttach(context);

        if (context instanceof NoteListListener) {
            noteListListener = (NoteListListener) context;
        } else {
            throw new RuntimeException(context.toString()
                    + " does not implement " + noteListListener.getClass().getName());
        }
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_note_list, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {

        floatingActionButton = view.findViewById(R.id.button_add_note);
        noteListRecyclerView = view.findViewById(R.id.note_list_recycler_view);

        floatingActionButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                noteListListener.startNewNote();
            }
        });

        noteViewModel = ViewModelProviders.of(Objects.requireNonNull(getActivity())).get(NoteViewModel.class);
        LiveData<List<NoteEntity>> notes = noteViewModel.getNotes();

        final NoteListAdapter noteListAdapter = new NoteListAdapter(notes.getValue(), this);
        noteListRecyclerView.setAdapter(noteListAdapter);
        noteListRecyclerView.setLayoutManager(new LinearLayoutManager(view.getContext()));

        notes.observe(this, new Observer<List<NoteEntity>>() {
            @Override
            public void onChanged(List<NoteEntity> noteEntities) {
                noteListAdapter.setNotes(noteEntities);
            }
        });
    }


    @Override
    public void onItemClick(int noteId) {
        noteListListener.editNote(noteId);
    }


}

File: app/src/main/java/com/example/user/biblejournal/view/notelist/NoteListFragment.kt
package com.example.user.biblejournal.view.notelist

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProviders
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

import com.example.user.biblejournal.R
import com.example.user.biblejournal.viewmodel.BibleViewModel
import com.google.android.material.floatingactionbutton.FloatingActionButton
import java.util.Objects


class NoteListFragment : Fragment(), NoteListAdapter.ListItemListener {
    private var noteListListener: NoteListListener? = null
    private var floatingActionButton: FloatingActionButton? = null
    private var bibleViewModel: BibleViewModel? = null
    private var noteListRecyclerView: RecyclerView? = null

    interface NoteListListener {
        fun startEditNote(noteId: Int?)
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)

        if (context is NoteListListener) {
            noteListListener = context
        } else {
            throw RuntimeException(context.toString()
                    + " does not implement " + noteListListener!!.javaClass.name)
        }
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_note_list, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {

        floatingActionButton = view.findViewById(R.id.button_add_note)
        noteListRecyclerView = view.findViewById(R.id.note_list_recycler_view)

        floatingActionButton!!.setOnClickListener { noteListListener!!.startEditNote(null) }

        bibleViewModel = ViewModelProviders.of(Objects.requireNonNull<FragmentActivity>(activity)).get(BibleViewModel::class.java)
        val notes = bibleViewModel!!.allNotes

        val noteListAdapter = NoteListAdapter(notes.value!!, this)
        noteListRecyclerView!!.adapter = noteListAdapter
        noteListRecyclerView!!.layoutManager = LinearLayoutManager(view.context)

        notes.observe(this, Observer { noteEntities -> noteListAdapter.setNotes(noteEntities) })

        bibleViewModel!!.readAllNotes()
    }

    override fun onItemClick(noteId: Int) {
        noteListListener!!.startEditNote(noteId)
    }

    companion object {

        fun newInstance(): NoteListFragment {
            return NoteListFragment()
        }
    }


}
